### PR TITLE
ci: removal of go actions and zbctl usage

### LINKED ISSUE #22511: Remove go client ci
```[tasklist]
### Acceptance Criteria
- [x] Remove Go jobs from `ci.yml`
- [x] Remove Go related setup from `setup-zeebe` action
- [x] Remove Go related setup from `build-zeebe` action
- [x] Adopt `zeebe-testbench` workflow to download zbctl instead
```

### PR DESCRIPTION
**Pull request by megglos**
## Description

This PR removes the go setup and build from our custom actions, as well as the related parameters on the action.

As the testbench workflow from is using zbctl it needs to be downloaded instead.

As a follow-up, once testbench is updated to 8.6.x we can start using the equivalent REST API to create a process instance, see https://github.com/camunda/camunda/issues/22536 .

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #22511

### CODE REVIEW

File: .github/actions/build-zeebe/action.yml [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
 ---
 name: Build Zeebe
 description: Builds & installs the complete Zeebe distribution
 
 inputs:
-  go:
-    description: If false, will not build zbctl; defaults to true
-    default: "true"
-    required: false
-  java:
```
Comment (megglos) [inline .github/actions/build-zeebe/action.yml:14]: I figured there is no usage for this param anymore, given this only builds java now

File: .github/workflows/zeebe-testbench.yaml [hunk 2]
```diff
@@ -80,16 +80,27 @@ jobs:
...
+            -H 'Accept: application/json' \
+            --data-raw '{
+              "bpmnProcessId": "${variables}",
+              "variables": $variables
+            }'
         env:
-          IMAGE: ${{ steps.build-zeebe-docker.outputs.image }}
           ZEEBE_CLIENT_SECRET: ${{ steps.secrets.outputs.TESTBENCH_PROD_CLIENT_SECRET }}
-          ZEEBE_ADDRESS: ${{ steps.secrets.outputs.TESTBENCH_PROD_CONTACT_POINT }}
+          ZEEBE_ADDRESS: ${{ steps.secrets.outputs.TESTBENCH_PROD_REST_ADDRESS }}
```
Comment (megglos) [inline .github/workflows/zeebe-testbench.yaml:104]: created this in the vault

**Comment by megglos** [comment]
Testrun https://github.com/camunda/camunda/actions/runs/10962451724

File: .ci/scripts/distribution/qa-testbench.sh [hunk 3]
```diff
@@ -20,7 +19,7 @@ jobKey=""
 
 while true; do
     # activate jobs non-zero exits can be ignored with `<command> || true`
-    "${zbctl}" activate jobs "$businessKey" > activationresponse.txt 2>/dev/null || true
+    zbctl activate jobs "$businessKey" > activationresponse.txt 2>/dev/null || true
```
Comment (Zelldon) [inline .ci/scripts/distribution/qa-testbench.sh:22]: Does this work? I would expect we need ./zbctl

File: .github/actions/build-zeebe/action.yml (hunk 1 above)
Comment (Zelldon) [reply .github/actions/build-zeebe/action.yml:14]: We need to adjust the workflow usage then

**Review by ChrisKujawa — APPROVED** [review]
👍

File: .ci/scripts/distribution/qa-testbench.sh (hunk 3 above)
Comment (megglos) [reply .ci/scripts/distribution/qa-testbench.sh]: fixed, my test run failed because of this :)

File: .github/actions/build-zeebe/action.yml (hunk 1 above)
Comment (megglos) [reply .github/actions/build-zeebe/action.yml:14]: the only ones using it were actually to go jobs which are already gone

**Pull request merged**